polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -464,15 +476,15 @@ async fn distribute_collation<Context>(
 		state.collation_result_senders.insert(candidate_hash, result_sender);
 	}
 
+	let parent_head_data = if elastic_scaling {
+		ParentHeadData::WithData { hash: parent_head_data_hash, head_data: parent_head_data }
```

Comment:
Why would we want to send the hash alongside the actual data? The validator will need to hash it itself anyways, to make sure it is correct.